pkgs/by-name/wa/wazuh/package.nix

```diff
@@ -0,0 +1,249 @@
+        sha256 = "sha256-iCeNS9/BvWo6GlWk89kzaD0nMroJz3p0n+jsjuxAbjw=";
+      })
+    (fetcher
+      {
+        name = "pacman";
+        sha256 = "sha256-9n3Tiir7NA19YDUNSbdamDp8TgGtdgIFaSDv6EnVsUM=";
+      })
+    (fetcher
+      {
+        name = "libarchive";
+        sha256 = "sha256-+GPzgurZ9hq8Vg/w6tC+OqnpW2+MYnVuHwNPTCOGunk=";
+      })
+    (fetcher
+      {
+        name = "popt";
+        sha256 = "sha256-1ogKBmIsoy3EqjmtXc977y+qgb2TGvvmS6Q0rY/uHao=";
+      })
+    (fetcher
+      {
+        name = "rpm";
+        sha256 = "sha256-TLxfyjwT3y3RUBLGi81SMH0omhvn6TxkjZ4P44ezFnA=";
+      })
+    (fetcher
+      {
+        name = "cpython";
+        sha256 = "sha256-Ct2yYX/g3PnnVvt9gOb1Uffqb/DCLIfcP8rTBva6yrA=";
+      })
+    (fetcher
+      {
+        name = "jemalloc";
+        sha256 = "sha256-KyLoWzUsffVQukCKQiUeUejf+myRqi4ftIBKsxf/vKA=";
+      })
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "wazuh-agent-build";
+  version = "4.7.3";
+
+  src = fetchFromGitHub {
+    owner = "wazuh";
+    repo = "wazuh";
+    rev = "v${version}";
+    sha256 = "sha256-h9gMAjE09f6nQ9zO7wdC+fspLKE6z5tlypHzdObJRGA=";
+  };
+
+  buildInputs = [
+    autoconf
+    automake
+    cmake
+    coreutils
+    curl
+    gcc
+    gnumake
+    libcxx
+    libtool
+    openssl
+    perl
+    policycoreutils
+    python312
+    zstd
+  ];
+
+  unpackPhase = with lib.strings; ''
+    mkdir -p $out/src/src/external
+    cp --no-preserve=all -rf $src/* $out/src
+    ${(concatStringsSep "\n" (map(dep: "tar -xzf ${dep} -C $out/src/src/external") dependencies))}
+  '';
+
+  patchPhase = ''
+    # Patch audit_userspace autogen.sh script
+    substituteInPlace $out/src/src/external/audit-userspace/autogen.sh \
+      --replace "cp INSTALL.tmp INSTALL" ""
+
+    # Required for OpenSSL to compile
+    # Patch the script to use the Nix store path directly for Perl
+    substituteInPlace $out/src/src/external/openssl/config \
+      --replace "/usr/bin/env" "env"
+
+    # Bypass check for tar file
+    touch $out/src/src/external/cpython.tar
+
+    cat << EOF > "$out/src/etc/preloaded-vars.conf"
+    USER_LANGUAGE="en"
+    USER_NO_STOP="y"
+    USER_INSTALL_TYPE="agent"
+    USER_DIR="$out"
+    USER_DELETE_DIR="n"
+    USER_ENABLE_ACTIVE_RESPONSE="y"
+    USER_ENABLE_SYSCHECK="y"
+    USER_ENABLE_ROOTCHECK="y"
+    USER_ENABLE_OPENSCAP="y"
+    USER_ENABLE_SYSCOLLECTOR="y"
+    USER_ENABLE_SECURITY_CONFIGURATION_ASSESSMENT="y"
+    USER_AGENT_SERVER_IP=127.0.0.1
+    USER_CA_STORE="no"
+    EOF
+
+    ln -sf ${libgcc.lib}/lib/libgcc_s.so.1 $out/src/src/libgcc_s.so.1
+    ln -sf ${libgcc.lib}/lib/libstdc++.so.6 $out/src/src/libstdc++.so.6
+  '';
+
+  configurePhase = ":";
+
+  buildPhase = ''
+    echo "Entering install phase"
+
+    echo "Running make deps..."
+    make -C $out/src/src deps
+
+    echo "Running make TARGET=agent..."
+    export OSSEC_LIBS="-lzstd"
+    make -C $out/src/src -j $(nproc) -d TARGET=agent INSTALLDIR=$out build
+  '';
+
+  fixupPhase = ":";
```

Comment:
What's this for...?

Reply:
Not sure, it was left in from previous work. I'll test without it, if the phase goes through without any bash instructions like this.

Reply:
> Not sure, it was left in from previous work. I'll test without it, if the phase goes through without any bash instructions like this.

My gut feeling would be that these were used to skip over certain phases, but nowadays that should be done by setting `doCheck`/`doBuild`/... to false.

Reply:
That makes sense! I'll look into those options then, thank you